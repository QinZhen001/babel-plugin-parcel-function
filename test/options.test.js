const {testPlugin} = require("./util/util")

const code = `function f(aa,bb) {
    let asd = 1 
    asd = aa + bb
    console.log("aaa")
  }
`


describe("测试插件参数", () => {
  test("参数为数组", () => {
    const result = testPlugin(code, [], [], {
      prefixName: ["aaa", "bbb", "ccc", "ddd"]
    })
    expect(result).toMatchSnapshot();
  })
  test("参数字符串", () => {
    const result = testPlugin(code, [], [], {
      prefixName: "aaa.bbb.ccc"
    })
    expect(result).toMatchSnapshot();
  })
  test("addFileName", () => {
    const result = testPlugin(code, [], [], {
      addFileName: true
    })
    expect(result).toMatchSnapshot();
  })
})


