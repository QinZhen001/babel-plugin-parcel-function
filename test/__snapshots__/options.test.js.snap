// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试插件参数 addFileName 1`] = `
"function f(aa, bb) {
    if (global.hotUpdate.f) {
        global.hotUpdate.f.call(this, aa, bb);
    } else {
        let asd = 1;
        asd = aa + bb;
        console.log(\\"aaa\\");
    }
}"
`;

exports[`测试插件参数 参数为数组 1`] = `
"function f(aa, bb) {
    if (aaa.bbb.ccc.ddd.f) {
        aaa.bbb.ccc.ddd.f.call(this, aa, bb);
    } else {
        let asd = 1;
        asd = aa + bb;
        console.log(\\"aaa\\");
    }
}"
`;

exports[`测试插件参数 参数字符串 1`] = `
"function f(aa, bb) {
    if (aaa.bbb.ccc.f) {
        aaa.bbb.ccc.f.call(this, aa, bb);
    } else {
        let asd = 1;
        asd = aa + bb;
        console.log(\\"aaa\\");
    }
}"
`;
